book/src/development/trait_checking.md

```diff
@@ -96,29 +96,36 @@ impl LateLintPass<'_> for CheckTokioAsyncReadExtTrait {
 
 ## Creating Types Programmatically
 
-Traits are often generic over a type e.g. `Borrow<T>` is generic over `T`, and rust allows us to implement a trait for
-a specific type. For example, we can implement `Borrow<str>` for a hypothetical type `Foo`. Let's suppose that we
-would like to find whether our type actually implements `Borrow<[u8]>`. To do so, we need to supply a type that
-represents `[u8]`, but `[u8]` is also a generic, it's a slice over `u8`. We can create this type using Ty::new_slice
-method. The following code demonstrates how to do this:
+Traits are often generic over a type parameter, e.g. `Borrow<T>` is generic
+over `T`. Rust allows us to implement a trait for a specific type. For example,
+we can implement `Borrow<[u8]>` for a hypothetical type `Foo`. Let's suppose
+that we would like to find whether our type actually implements `Borrow<[u8]>`.
+
+To do so, we can use the same `implements_trait` function as above, and supply
+a parameter type that represents `[u8]`. Since `[u8]` is a specialization of
+`[T]`, we can use the  [`Ty::new_slice`][new_slice] method to create a type
+that represents `[T]` and supply `u8` as a type parameter. The following code
```

Comment:
Should we mention that we have to turn this value into a generic first?

Reply:
I would add a `// <- turn into generic parameter` comment in the code below. Also point and comment the `new_slice` line in the code below.